servicetalk-http-api/src/main/java/io/servicetalk/http/api/Http2SettingsBuilder.java

```diff
@@ -0,0 +1,201 @@
+        settings.put(INITIAL_WINDOW_SIZE, value);
+        return this;
+    }
+
+    /**
+     * Set the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_MAX_FRAME_SIZE</a>.
+     *
+     * @param value The value.
+     * @return {@code this}.
+     */
+    public Http2SettingsBuilder maxFrameSize(int value) {
+        settings.put(MAX_FRAME_SIZE, value);
+        return this;
+    }
+
+    /**
+     * Set the value for
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.2">SETTINGS_MAX_HEADER_LIST_SIZE</a>.
+     *
+     * @param value The value.
+     * @return {@code this}.
+     */
+    public Http2SettingsBuilder maxHeaderListSize(int value) {
+        settings.put(MAX_HEADER_LIST_SIZE, value);
+        return this;
+    }
+
+    /**
+     * Build the {@link Map} that represents
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.1">HTTP/2 Setting</a>.
+     * @return the {@link Map} that represents
+     * <a href="https://datatracker.ietf.org/doc/html/rfc7540#section-6.5.1">HTTP/2 Setting</a>.
+     */
+    public Http2Settings build() {
+        return new DefaultHttp2Settings(settings);
+    }
+
+    private static final class DefaultHttp2Settings implements Http2Settings {
+        private final Map<Character, Integer> settings;
+
+        private DefaultHttp2Settings(final Map<Character, Integer> settings) {
+            this.settings = settings;
+        }
+
+        @Nullable
+        @Override
+        public Integer headerTableSize() {
+            return settings.get(HEADER_TABLE_SIZE);
+        }
+
+        @Nullable
+        @Override
+        public Integer maxConcurrentStreams() {
+            return settings.get(MAX_CONCURRENT_STREAMS);
+        }
+
+        @Nullable
+        @Override
+        public Integer initialWindowSize() {
+            return settings.get(INITIAL_WINDOW_SIZE);
+        }
+
+        @Nullable
+        @Override
+        public Integer maxFrameSize() {
+            return settings.get(MAX_FRAME_SIZE);
+        }
+
+        @Nullable
+        @Override
+        public Integer maxHeaderListSize() {
+            return settings.get(MAX_HEADER_LIST_SIZE);
+        }
+
+        @Nullable
+        @Override
+        public Integer settingValue(final char identifier) {
+            return settings.get(identifier);
+        }
+
+        @Override
+        public void forEach(final BiConsumer<? super Character, ? super Integer> action) {
+            settings.forEach(action);
+        }
+
+        @Override
+        public String toString() {
+            return settings.toString();
+        }
+
+        @Override
+        public int hashCode() {
+            return settings.hashCode();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            return o instanceof DefaultHttp2Settings && settings.equals(((DefaultHttp2Settings) o).settings);
```

Comment:
For the purpose of `equals` and `hashCode` it should not matter what is the impl type of the `Http2Settings` interface. We should compare only the actual values present.

Reply:
On the other hand, maybe we should. Otherwise, users can extend `Http2Settings` to add more characteristics. Let's keep the `instanceof`.